src/charm.py

```diff
@@ -80,11 +85,45 @@ def _on_config_changed(self, _):
         else:
             self.unit.status = WaitingStatus("Waiting for Pebble in workload container")
 
+    def _patch_auth_resources(self):
+        """Returns True if existing auth resources for training-operator are patched correctly."""
+        self.unit.status = MaintenanceStatus("Patching auth configuration")
+        client = Client()
+        context = {"namespace": self._namespace}
+
+        with open(os.path.join(self._src_dir, "auth_manifests.yaml")) as f:
+            for obj in codecs.load_all_yaml(f, context=context):
+                try:
+                    client.patch(
+                        type(obj), obj.metadata.name, obj, patch_type=PatchType.MERGE
+                    )
+                except ApiError as e:
+                    logging.error(f"Patching auth resources failed: {str(e)}")
+        logging.info("Auth resources successfully patched.")
+        return True
```

Comment:
Unless I'm missing something, in the case of an ApiError you would log an error but still end up returning True. Seems like there is a `return False` statement missing.

Reply:
Agree on the missing `return False`. Will work on it.

Reply:
Updated it in latest PR version. Resolving this.